### ISSUE TITLE
PyQgsServerWMSDimension modifies source file tests/testdata/qgis_server_accesscontrol/dimensions.gpkg

### ISSUE THREAD

**Issue by strk** [qgis/QGIS#49033]
### What is the bug or the crash?

```
Test project /usr/local/src/qgis/qgis/src/master/build
    Start   1: logGitStatus
1/3 Test   #1: logGitStatus .....................   Passed    0.07 sec
    Start 826: PyQgsServerWMSDimension
2/3 Test #826: PyQgsServerWMSDimension ..........   Passed   17.81 sec
    Start   2: checkGitStatus
3/3 Test   #2: checkGitStatus ...................***Failed    0.17 sec
1d0
<       modified:   tests/testdata/qgis_server_accesscontrol/dimensions.gpkg
```

### Steps to reproduce the issue

`xvfb-run ctest --output-on-failure -R PyQgsServerWMSDimension$`

### Versions

Commit 5973fb315cdb6d3b3c01e9a33b5339a00b9c9e4d

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

Spin-off of https://github.com/qgis/QGIS/issues/25830

Still happens after committing 7d3977a22e833159e518eafadd14b32123046fc0 and ea14d5f024bb6c653073f8c4784a1d5177245f43

**Comment by strk** [comment]
The line which triggers modification of the source file is this line:

> project = mConfigCache->project( configFilePath, sServerInterface->serverSettings() );

In QgsServer::handleRequest method ( src/server/qgsserver.cpp )

At that time (before the call to `project`) `qgis/walForSqlite3` is set to False

**Comment by strk** [comment]
More specifically, it is the QgsProject::read call that modifies the source file, readFlags are:

XXXX HANDLEREQUEST reading project, readFlags: QFlags<Qgis::ProjectReadFlag>(DontStoreOriginalStyles|DontLoad3DViews|DontLoadProjectStyles)


I run a diff of the sqlite3 schema before and after the introduced modification. Maybe you have some hint about this @rouault 

```
 CREATE TRIGGER "rtree_contours_geom_update3" AFTER UPDATE OF "geom" ON "contours" WHEN OLD."fid" != NEW."fid" AND (NEW."geom" NOTNULL AND NOT ST_IsEmpty(NEW."geom")) BEGIN DELETE FROM "rtree_contours_geom" WHERE id = OLD."fid"; INSERT OR REPLACE INTO "rtree_contours_geom" VALUES (NEW."fid",ST_MinX(NEW."geom"), ST_MaxX(NEW."geom"),ST_MinY(NEW."geom"), ST_MaxY(NEW."geom")); END;
55d53
< CREATE TRIGGER "rtree_slopes_geom_update3" AFTER UPDATE OF "geom" ON "slopes" WHEN OLD."fid" != NEW."fid" AND (NEW."geom" NOTNULL AND NOT ST_IsEmpty(NEW."geom")) BEGIN DELETE FROM "rtree_slopes_geom" WHERE id = OLD."fid"; INSERT OR REPLACE INTO "rtree_slopes_geom" VALUES (NEW."fid",ST_MinX(NEW."geom"), ST_MaxX(NEW."geom"),ST_MinY(NEW."geom"), ST_MaxY(NEW."geom")); END;
63a62,63
> CREATE TRIGGER "rtree_contours_geom_update3" AFTER UPDATE ON "contours" WHEN OLD."fid" != NEW."fid" AND (NEW."geom" NOTNULL AND NOT ST_IsEmpty(NEW."geom")) BEGIN DELETE FROM "rtree_contours_geom" WHERE id = OLD."fid"; INSERT OR REPLACE INTO "rtree_contours_geom" VALUES (NEW."fid",ST_MinX(NEW."geom"), ST_MaxX(NEW."geom"),ST_MinY(NEW."geom"), ST_MaxY(NEW."geom")); END;
> CREATE TRIGGER "rtree_slopes_geom_update3" AFTER UPDATE ON "slopes" WHEN OLD."fid" != NEW."fid" AND (NEW."geom" NOTNULL AND NOT ST_IsEmpty(NEW."geom")) BEGIN DELETE FROM "rtree_slopes_geom" WHERE id = OLD."fid"; INSERT OR REPLACE INTO "rtree_slopes_geom" VALUES (NEW."fid",ST_MinX(NEW."geom"), ST_MaxX(NEW."geom"),ST_MinY(NEW."geom"), ST_MaxY(NEW."geom")); END;
```

It looks to me that the modification drops the `OF "<fieldname>"` in the definition of the `rtree_contours_geom_update3` and `rtree_slopes_geom_update3` triggers. Does it ring any bell ? @elpaso to you ?

**Comment by rouault** [comment]
Since https://github.com/OSGeo/gdal/pull/2654, GDAL >= 3.2 fixes wrong triggers created by GDAL <= 2.3
The files in the repository should likely be updated with that fixup

**Comment by strk** [comment]
Thanks @rouault I'll try to do so.
For the record, GDAL-3.0.4 used by CI does not indeed modify that file. Locally I'm running 3.2.2 which does.
Is there a simple incantation to use on commandline to do a batch fixup of those triggers ?
